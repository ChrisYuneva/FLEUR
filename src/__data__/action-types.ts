export const GOODS = {
    REQUEST: 'GOODS_REQUEST',
    SUCCESS: 'GOODS_SUCCESS',
    FAILURE: 'GOODS_FAILURE',
    SET_GENDER: 'GOODS_SET_GENDER',
    SET_TYPE: 'GOODS_SET_TYPE',
};

export const BASKET = {
    REQUEST: 'BASKET_REQUEST',
    SUCCESS: 'BASKET_SUCCESS',
    FAILURE: 'BASKET_FAILURE',
};
