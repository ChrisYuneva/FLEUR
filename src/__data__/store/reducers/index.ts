import { combineReducers } from 'redux';
import goodsReducer, { GoodsState } from './goods';
import basketReducer, { BasketState } from './basket';

export type AppStore = {
    goods: GoodsState;
    basket: BasketState;
};

export default combineReducers<AppStore>({
    goods: goodsReducer,
    basket: basketReducer,
});
