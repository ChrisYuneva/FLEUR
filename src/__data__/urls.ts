import { getNavigations } from '@ijl/cli';

const navigations = getNavigations('repos');

export const baseUrl = navigations['repos'];
export const URLs = {
    basket: {
        url: navigations['link.repos.basket'],
    },
    home: {
        url: navigations['link.repos.home'],
    },
    home_man: {
        url: navigations['link.repos.home_man'],
    },
    showcase_men: {
        url: navigations['link.repos.showcase_men'],
    },
    showcase_women: {
        url: navigations['link.repos.showcase_women'],
    },
    ordering: {
        url: navigations['link.repos.ordering'],
    },
};
