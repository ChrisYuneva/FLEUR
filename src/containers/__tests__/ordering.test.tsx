import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Input from '../ordering/steps/input';
import Luck from '../ordering/steps/luck';
import Order from '../ordering/ordering';
import { describe, it, expect } from '@jest/globals';
import { store } from '../../__data__/store';

describe('Тестирование', () => {
    it('Тестируем рендер Input', () => {
        expect.assertions(1);
        const component = mount(
            <Provider store={store}>
                <Input nextStep={onclick} />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });

    it('Тестируем рендер Luck', () => {
        expect.assertions(1);
        const component = mount(
            <Provider store={store}>
                <Luck />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });

    it('Тестируем рендер Order', () => {
        expect.assertions(1);
        const component = mount(
            <Provider store={store}>
                <Order />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });
});
