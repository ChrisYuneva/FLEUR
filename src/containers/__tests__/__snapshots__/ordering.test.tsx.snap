// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование Тестируем рендер Input 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Input
    nextStep={null}
  >
    <div
      className="wrapper"
    >
      <header
        className="header"
      >
        <Link
          className="name"
          to="/home"
        >
          <a
            className="name"
            href="/home"
          >
            FLEUR
          </a>
        </Link>
      </header>
      <ReactFinalForm
        onSubmit={[Function]}
        render={[Function]}
      >
        <div
          className="order"
        >
          <h1>
            Оформление заказа
          </h1>
          <div
            className="form"
          >
            <div
              className="input"
            >
              <label>
                Введите имя
              </label>
              <ForwardRef(Field)
                component="input"
                name="firstName"
                type="text"
              >
                <input
                  name="firstName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text"
                  value=""
                />
              </ForwardRef(Field)>
            </div>
            <div
              className="input"
            >
              <label>
                Введите фамилию
              </label>
              <ForwardRef(Field)
                component="input"
                name="lastName"
                type="text"
              >
                <input
                  name="lastName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text"
                  value=""
                />
              </ForwardRef(Field)>
            </div>
            <div
              className="input"
            >
              <label>
                Введите адрес
              </label>
              <ForwardRef(Field)
                component="input"
                name="adress"
                type="text"
              >
                <input
                  name="adress"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text"
                  value=""
                />
              </ForwardRef(Field)>
            </div>
            <div
              className="input"
            >
              <label>
                Введите почтовый индекс
              </label>
              <ForwardRef(Field)
                component="input"
                name="index"
                type="text"
              >
                <input
                  name="index"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text"
                  value=""
                />
              </ForwardRef(Field)>
            </div>
            <div />
            <button
              className="arrange"
              onClick={null}
            >
              Оформить
            </button>
          </div>
        </div>
      </ReactFinalForm>
    </div>
  </Input>
</Provider>
`;

exports[`Тестирование Тестируем рендер Luck 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Luck>
    <div
      className="orderS"
    >
      <h1
        className="luck"
      >
        Ваш заказ оформлен!
      </h1>
      <Link
        className="continue"
        to="/showcase_dress"
      >
        <a
          className="continue"
          href="/showcase_dress"
        >
          Продолжить покупки
        </a>
      </Link>
    </div>
  </Luck>
</Provider>
`;

exports[`Тестирование Тестируем рендер Order 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Order>
    <div>
      <Input
        nextStep={[Function]}
      >
        <div
          className="wrapper"
        >
          <header
            className="header"
          >
            <Link
              className="name"
              to="/home"
            >
              <a
                className="name"
                href="/home"
              >
                FLEUR
              </a>
            </Link>
          </header>
          <ReactFinalForm
            onSubmit={[Function]}
            render={[Function]}
          >
            <div
              className="order"
            >
              <h1>
                Оформление заказа
              </h1>
              <div
                className="form"
              >
                <div
                  className="input"
                >
                  <label>
                    Введите имя
                  </label>
                  <ForwardRef(Field)
                    component="input"
                    name="firstName"
                    type="text"
                  >
                    <input
                      name="firstName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="text"
                      value=""
                    />
                  </ForwardRef(Field)>
                </div>
                <div
                  className="input"
                >
                  <label>
                    Введите фамилию
                  </label>
                  <ForwardRef(Field)
                    component="input"
                    name="lastName"
                    type="text"
                  >
                    <input
                      name="lastName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="text"
                      value=""
                    />
                  </ForwardRef(Field)>
                </div>
                <div
                  className="input"
                >
                  <label>
                    Введите адрес
                  </label>
                  <ForwardRef(Field)
                    component="input"
                    name="adress"
                    type="text"
                  >
                    <input
                      name="adress"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="text"
                      value=""
                    />
                  </ForwardRef(Field)>
                </div>
                <div
                  className="input"
                >
                  <label>
                    Введите почтовый индекс
                  </label>
                  <ForwardRef(Field)
                    component="input"
                    name="index"
                    type="text"
                  >
                    <input
                      name="index"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="text"
                      value=""
                    />
                  </ForwardRef(Field)>
                </div>
                <div />
                <button
                  className="arrange"
                  onClick={[Function]}
                >
                  Оформить
                </button>
              </div>
            </div>
          </ReactFinalForm>
        </div>
      </Input>
    </div>
  </Order>
</Provider>
`;
